### ISSUE TITLE
kola: rawhide: 20211213: `ext.config.podman.dns` test failing 

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1046]
**Describe the bug**
The `ext.config.podman.dns` started failing in `rawhide` [link](https://jenkins-fedora-coreos.apps.ocp.ci.centos.org/blue/organizations/jenkins/fedora-coreos%2Ffedora-coreos-fedora-coreos-pipeline/detail/fedora-coreos-fedora-coreos-pipeline/3414/pipeline/150). The full error can be seen by running the test manually:

```
[core@cosa-devsh ~]$ podman network create testnetwork
/var/home/core/.config/cni/net.d/testnetwork.conflist
[core@cosa-devsh ~]$ podman run --rm -t --network=testnetwork registry.fedoraproject.org/fedora:35 getent hosts google.com
ERRO[0000] error loading cached network config: network "testnetwork" not found in CNI cache 
WARN[0000] falling back to loading from existing plugins on disk 
ERRO[0000] Error tearing down partially created network namespace for container 539c41e10d9b5b62c020b71ed73891bf97c257e974928507dd1b967b33cb0125: error removing pod practical_lamport_practical_lamport from CNI network "testnetwork": running [/usr/sbin/iptables -t nat -D POSTROUTING -s 10.89.0.2 -j CNI-bd2745568755c29ef2bf4f1a -m comment --comment name: "testnetwork" id: "539c41e10d9b5b62c020b71ed73891bf97c257e974928507dd1b967b33cb0125" --wait]: exit status 3: modprobe: ERROR: could not insert 'ip_tables': Operation not permitted
                                                               iptables v1.8.7 (legacy): can't initialize iptables table `nat': Table does not exist (do you need to insmod?)
                Perhaps iptables or your kernel needs to be upgraded. 
Error: error configuring network namespace for container 539c41e10d9b5b62c020b71ed73891bf97c257e974928507dd1b967b33cb0125: error adding pod practical_lamport_practical_lamport to CNI network "testnetwork": failed to list chains: running [/usr/sbin/iptables -t nat -S --wait]: exit status 3: modprobe: ERROR: could not insert 'ip_tables': Operation not permitted
iptables v1.8.7 (legacy): can't initialize iptables table `nat': Table does not exist (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.
```

It appears it's having trouble loading the `ip_tables` kernel module.

This list of changed packages since the last successful run:

```
[2021-12-13T18:49:20.385Z] Upgraded:
[2021-12-13T18:49:20.385Z]   console-login-helper-messages 0.21.2-2.fc35 -> 0.21.2-3.fc36
[2021-12-13T18:49:20.385Z]   console-login-helper-messages-issuegen 0.21.2-2.fc35 -> 0.21.2-3.fc36
[2021-12-13T18:49:20.385Z]   console-login-helper-messages-motdgen 0.21.2-2.fc35 -> 0.21.2-3.fc36
[2021-12-13T18:49:20.385Z]   console-login-helper-messages-profile 0.21.2-2.fc35 -> 0.21.2-3.fc36
[2021-12-13T18:49:20.385Z]   glibc 2.34.9000-26.fc36 -> 2.34.9000-27.fc36
[2021-12-13T18:49:20.385Z]   glibc-common 2.34.9000-26.fc36 -> 2.34.9000-27.fc36
[2021-12-13T18:49:20.385Z]   glibc-minimal-langpack 2.34.9000-26.fc36 -> 2.34.9000-27.fc36
[2021-12-13T18:49:20.385Z]   grub2-common 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   grub2-efi-x64 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   grub2-pc 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   grub2-pc-modules 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   grub2-tools 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   grub2-tools-minimal 1:2.06-9.fc36 -> 1:2.06-11.fc36
[2021-12-13T18:49:20.385Z]   libsmbclient 2:4.15.2-3.fc36 -> 2:4.15.3-0.fc36
[2021-12-13T18:49:20.385Z]   libwbclient 2:4.15.2-3.fc36 -> 2:4.15.3-0.fc36
[2021-12-13T18:49:20.385Z]   ostree 2021.6-1.fc36 -> 2021.6-3.fc36
[2021-12-13T18:49:20.385Z]   ostree-libs 2021.6-1.fc36 -> 2021.6-3.fc36
[2021-12-13T18:49:20.385Z]   samba-client-libs 2:4.15.2-3.fc36 -> 2:4.15.3-0.fc36
[2021-12-13T18:49:20.385Z]   samba-common 2:4.15.2-3.fc36 -> 2:4.15.3-0.fc36
[2021-12-13T18:49:20.385Z]   samba-common-libs 2:4.15.2-3.fc36 -> 2:4.15.3-0.fc36
[2021-12-13T18:49:20.385Z]   systemd 249.7-2.fc36 -> 250~rc1-2.fc36
[2021-12-13T18:49:20.385Z]   systemd-container 249.7-2.fc36 -> 250~rc1-2.fc36
[2021-12-13T18:49:20.385Z]   systemd-libs 249.7-2.fc36 -> 250~rc1-2.fc36
[2021-12-13T18:49:20.385Z]   systemd-pam 249.7-2.fc36 -> 250~rc1-2.fc36
[2021-12-13T18:49:20.385Z]   systemd-resolved 249.7-2.fc36 -> 250~rc1-2.fc36
[2021-12-13T18:49:20.385Z]   systemd-udev 249.7-2.fc36 -> 250~rc1-2.fc36
```

**Comment by dustymabe** [comment]
pin over in https://github.com/coreos/fedora-coreos-config/pull/1370

**Comment by miabbott** [comment]
I scanned the release notes for `systemd-250` but nothing jumped out - https://lists.freedesktop.org/archives/systemd-devel/2021-December/047170.html

Perhaps someone with a more critical eye sees something suspect

**Comment by travier** [comment]
I'm trying to build FCOS Rawhide with the systemd pin reverted and I get:

```
Initializing rootfs
Moving /usr to target
Copying toplevel compat symlinks
Recompiling policy
Migrating /usr/etc/passwd to /usr/lib/
Migrating /usr/etc/group to /usr/lib/
error: Finalizing rootfs: Adding altfiles to /etc/nsswitch.conf: No such file or directory (os error 2)
```

**Comment by dustymabe** [comment]
Thanks for looking at this @travier.

**Comment by travier** [comment]
See also https://github.com/coreos/fedora-coreos-config/pull/1376

**Comment by travier** [comment]
Pending figuring out how to build latest FCOS Rawhide, I've ran the manually on a Kinoite Rawhide host with systemd 250-rc1 and it worked so there must be something else going on.

**Comment by lucab** [comment]
I didn't dig into this on a booted system, but I wouldn't expect an **unprivileged** podman execution to be able to load a module.
So possibly, something was auto-loading it for us before and it isn't doing that anymore.

Looking around, I do feel that may have been systemd [itself](
https://github.com/systemd/systemd/blob/f6278558da0304ec6b646bb172ce4688c7f162a5/src/core/kmod-setup.c#L88-L91):
```
#if HAVE_LIBIPTC
                /* netfilter is needed by networkd, nspawn among others, and cannot be autoloaded */
                { "ip_tables", "/proc/net/ip_tables_names", false,  false,   NULL      },
#endif
```

And the `249.7-3` [update](https://src.fedoraproject.org/rpms/systemd/pull-request/68) may have turned that off:
```
[...]
-        -Dlibiptc=true
+        -Dlibiptc=false
[...]
 %changelog
+* Fri Nov 19 2021 Davide Cavalca <dcavalca@fedoraproject.org> - 249.7-3
+- Disable legacy iptables support
```

**Comment by cgwalters** [comment]
> but I wouldn't expect an unprivileged podman execution to be able to load a module.

It is actually a totally normal thing for arbitrary userspace processes to be able to queue a module load: https://lwn.net/Articles/740455/

Though here it seems more that iptables isn't supported to be auto-loaded, but systemd was always loading it by default.  And podman was relying on that implicitly.

So...this is kind of messy because iptables is only necessary I would assume for *some* uses of podman.  

I guess though we could for now unconditionally load it.  But, it will be a trap for other distributions and podman (and I assume docker) too.

**Comment by lucab** [comment]
Yes but it looks like something in this unprivileged context is actively trying to modprobe the module.
The actual error from the top post in this ticket is:
```
Error tearing down [...] exit status 3: modprobe: ERROR: could not insert 'ip_tables': Operation not permitted
```

**Comment by lucab** [comment]
Reading again the error log, I got some doubts that this teardown logic may not be actually expected to run in this unprivileged context. I asked the podman folks to have a look at this flow at https://github.com/containers/podman/issues/12661.

**Comment by dustymabe** [comment]
Fix landed upstream in https://github.com/containers/podman/commit/113f855802176a4ffbe167304ab83942116a65f1

**Comment by dustymabe** [comment]
Should be fixed in podman-4.0.1-2 and newer RPMs. See https://github.com/containers/podman/pull/13323#issuecomment-1054432166

**Comment by travier** [comment]
`next` directly shipped with podman 4.0.2 thus this never affected a release.
